numpy/lib/function_base.py

```diff
@@ -1299,7 +1299,6 @@ def piecewise(x, condlist, funclist, *args, **kw):
                                    isinstance(condlist[0], ndarray))):
         condlist = [condlist]
     condlist = array(condlist, dtype=bool)
-    n = len(condlist)
     # This is a hack to work around problems with NumPy's
     #  handling of 0-d arrays and boolean indexing with
     #  numpy.bool_ scalars
```

Comment:
Do you mean the term "hack" in line 1303?
If so, the bug that I pointed out is not yet solved without my patch.  The comment is for another problem.